### ISSUE TITLE
Problem: package docker-ce-3:20.10.1-3.el8.x86_64 requires docker-ce-rootless-extras, but none of the providers can be installed

### ISSUE THREAD

**Issue by varyumin** [geerlingguy/ansible-role-docker#247]
Hello! I have some problem. I have used this role  "geerlingguy.docker" in my playbook and tried to install docker on clean Centos 8

**OS version:**
```
CentOS Linux release 8.3.2011
Derived from Red Hat Enterprise Linux 8.3
NAME="CentOS Linux"
VERSION="8"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Linux 8"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"
CENTOS_MANTISBT_PROJECT="CentOS-8"
CENTOS_MANTISBT_PROJECT_VERSION="8"
CentOS Linux release 8.3.2011
CentOS Linux release 8.3.2011
cpe:/o:centos:centos:8
```

**YUM version:**
```
4.2.23
  Installed: dnf-0:4.2.23-4.el8.noarch at Thu Dec 10 14:41:40 2020
  Built    : CentOS Buildsys <bugs@centos.org> at Tue Aug  4 18:52:03 2020

  Installed: rpm-0:4.14.3-4.el8.x86_64 at Thu Dec 10 14:41:17 2020
  Built    : CentOS Buildsys <bugs@centos.org> at Tue Jul 21 17:36:08 2020
```

**Galaxy install:**
```
Starting galaxy role install process
- changing role geerlingguy.docker from 2.9.0 to unspecified
- downloading role 'docker', owned by geerlingguy
- downloading role from https://github.com/geerlingguy/ansible-role-docker/archive/2.9.0.tar.gz
- extracting geerlingguy.docker to /mnt/d/Git/production/ansible/roles/geerlingguy.docker
- geerlingguy.docker (2.9.0) was installed successfully
```

**Run playbook**
```
PLAY [vm] **************************************************************************************************************************************************************************************************************************************************************

TASK [geerlingguy.docker : include_tasks] ************************************************************************************************************************************************************************************************************************************
included: /mnt/d/Git/production/ansible/roles/geerlingguy.docker/tasks/setup-RedHat.yml for cpx-teamcity-01

TASK [geerlingguy.docker : Ensure old versions of Docker are not installed.] *************************************************************************************************************************************************************************************************
ok: [vm]

TASK [geerlingguy.docker : Add Docker GPG key.] ******************************************************************************************************************************************************************************************************************************
[WARNING]: Module remote_tmp /root/.ansible/tmp did not exist and was created with a mode of 0700, this may cause issues when running as another user. To avoid this, create the remote_tmp dir with the correct permissions manually
changed: [vm]

TASK [geerlingguy.docker : Add Docker repository.] ***************************************************************************************************************************************************************************************************************************
changed: [vm]

TASK [geerlingguy.docker : Configure Docker Edge repo.] **********************************************************************************************************************************************************************************************************************
changed: [vm]

TASK [geerlingguy.docker : Configure Docker Test repo.] **********************************************************************************************************************************************************************************************************************
changed: [vm]

TASK [geerlingguy.docker : Ensure container-selinux is installed.] ***********************************************************************************************************************************************************************************************************
changed: [vm]

TASK [geerlingguy.docker : Disable container-tools module.] ******************************************************************************************************************************************************************************************************************
ok: [vm]

TASK [geerlingguy.docker : Ensure containerd.io is installed.] ***************************************************************************************************************************************************************************************************************
changed: [vm]

TASK [geerlingguy.docker : include_tasks] ************************************************************************************************************************************************************************************************************************************

TASK [geerlingguy.docker : Install Docker.] **********************************************************************************************************************************************************************************************************************************
fatal: [vm]: FAILED! => {"changed": false, "failures": [], "msg": "Depsolve Error occured: \n Problem: package docker-ce-3:20.10.1-3.el8.x86_64 requires docker-ce-rootless-extras, but none of the providers can be installed\n  - package docker-ce-rootless-extras-20.10.0-3.el8.x86_64 requires slirp4netns >= 0.4, but none of the providers can be installed\n  - package docker-ce-rootless-extras-20.10.1-3.el8.x86_64 requires slirp4netns >= 0.4, but none of the providers can be installed\n  - cannot install the best candidate for the job\n  - package slirp4netns-0.4.2-3.git21fdece.module_el8.3.0+479+69e2ae26.x86_64 is filtered out by modular filtering\n  - package slirp4netns-1.1.4-2.module_el8.3.0+475+c50ce30b.x86_64 is filtered out by modular filtering", "rc": 1, "results": []}

PLAY RECAP *******************************************************************************************************************************************************************************************************************************************************************
vm            : ok=9    changed=6    unreachable=0    failed=1    skipped=1    rescued=0    ignored=0
```
**/etc/yum.repos.d/docker-ce.repo**:
```
[docker-ce-stable]
name=Docker CE Stable - $basearch
baseurl=https://download.docker.com/linux/centos/$releasever/$basearch/stable
enabled=1
gpgcheck=1
gpgkey=https://download.docker.com/linux/centos/gpg

[docker-ce-stable-debuginfo]
name=Docker CE Stable - Debuginfo $basearch
baseurl=https://download.docker.com/linux/centos/$releasever/debug-$basearch/stable
enabled=0
gpgcheck=1
gpgkey=https://download.docker.com/linux/centos/gpg

[docker-ce-stable-source]
name=Docker CE Stable - Sources
baseurl=https://download.docker.com/linux/centos/$releasever/source/stable
enabled=0
gpgcheck=1
gpgkey=https://download.docker.com/linux/centos/gpg

[docker-ce-test]
name=Docker CE Test - $basearch
baseurl=https://download.docker.com/linux/centos/$releasever/$basearch/test
enabled = 0
gpgcheck=1
gpgkey=https://download.docker.com/linux/centos/gpg

[docker-ce-test-debuginfo]
name=Docker CE Test - Debuginfo $basearch
baseurl=https://download.docker.com/linux/centos/$releasever/debug-$basearch/test
enabled=0
gpgcheck=1
gpgkey=https://download.docker.com/linux/centos/gpg

[docker-ce-test-source]
name=Docker CE Test - Sources
baseurl=https://download.docker.com/linux/centos/$releasever/source/test
enabled=0
gpgcheck=1
gpgkey=https://download.docker.com/linux/centos/gpg

[docker-ce-nightly]
name=Docker CE Nightly - $basearch
baseurl=https://download.docker.com/linux/centos/$releasever/$basearch/nightly
enabled=0
gpgcheck=1
gpgkey=https://download.docker.com/linux/centos/gpg

[docker-ce-nightly-debuginfo]
name=Docker CE Nightly - Debuginfo $basearch
baseurl=https://download.docker.com/linux/centos/$releasever/debug-$basearch/nightly
enabled=0
gpgcheck=1
gpgkey=https://download.docker.com/linux/centos/gpg

[docker-ce-nightly-source]
name=Docker CE Nightly - Sources
baseurl=https://download.docker.com/linux/centos/$releasever/source/nightly
enabled=0
gpgcheck=1
gpgkey=https://download.docker.com/linux/centos/gpg
[docker-ce-edge]
enabled = 0
```

Can somebody help me? How I can fix it?

**Comment by varyumin** [comment]
I tried to install it on a remote server using yum```
yum install docker-ce
Repository 'docker-ce-edge' is missing name in configuration, using id.
Last metadata expiration check: 0:36:26 ago on Thu Dec 24 14:00:32 2020.
Error:
 Problem: package docker-ce-3:20.10.1-3.el8.x86_64 requires docker-ce-rootless-extras, but none of the providers can be installed
  - package docker-ce-rootless-extras-20.10.0-3.el8.x86_64 requires slirp4netns >= 0.4, but none of the providers can be installed
  - package docker-ce-rootless-extras-20.10.1-3.el8.x86_64 requires slirp4netns >= 0.4, but none of the providers can be installed
  - cannot install the best candidate for the job
  - package slirp4netns-0.4.2-3.git21fdece.module_el8.3.0+479+69e2ae26.x86_64 is filtered out by modular filtering
  - package slirp4netns-1.1.4-2.module_el8.3.0+475+c50ce30b.x86_64 is filtered out by modular filtering
(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
```

**Comment by varyumin** [comment]
https://github.com/docker/docker-install/issues/204

**Comment by geerlingguy** [comment]
Note that the install is currently failing due to https://github.com/geerlingguy/ansible-role-docker/issues/243 — I just submitted a PR that might fix it: https://github.com/geerlingguy/ansible-role-docker/pull/248

**Comment by varyumin** [comment]
It fixed in version 3.0.0. I closed this issue! Thx! You are the best!
